src/pkg/services/m365/api/channels.go

```diff
@@ -1 +1,376 @@
+func (c Channels) GetChannel(
+	ctx context.Context,
+	teamID, containerID string,
+) (models.Channelable, error) {
+	config := &teams.ItemChannelsChannelItemRequestBuilderGetRequestConfiguration{
+		QueryParameters: &teams.ItemChannelsChannelItemRequestBuilderGetQueryParameters{
+			Select: idAnd("name", "owner"),
+		},
+	}
+
+	resp, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Get(ctx, config)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return resp, nil
+}
+
+func (c Channels) GetChannelByID(
+	ctx context.Context,
+	teamID, containerID string,
+) (graph.Container, error) {
+	channel, err := c.GetChannel(ctx, teamID, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	return ChannelsDisplayable{Channelable: channel}, nil
+}
+
+// GetChannelByName fetches a calendar by name
+func (c Channels) GetChannelByName(
+	ctx context.Context,
+	teamID, _, containerName string,
+) (graph.Container, error) {
+
+	ctx = clues.Add(ctx, "channel_name", containerName)
+
+	resp, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		Get(ctx, nil)
+
+	if err != nil {
+		return nil, graph.Stack(ctx, err).WithClues(ctx)
+	}
+
+	gv := resp.GetValue()
+
+	if len(gv) == 0 {
+		return nil, clues.New("channel not found").WithClues(ctx)
+	}
+
+	// We only allow the api to match one calendar with the provided name.
+	// If we match multiples, we'll eagerly return the first one.
+	logger.Ctx(ctx).Debugw("calendars matched the name search", "calendar_count", len(gv))
+
+	// Sanity check ID and name
+	cal := gv[0]
+	container := ChannelsDisplayable{Channelable: cal}
+
+	if err := graph.CheckIDAndName(container); err != nil {
+		return nil, clues.Stack(err).WithClues(ctx)
+	}
+
+	return container, nil
+}
+
+func (c Channels) PatchChannel(
+	ctx context.Context,
+	teamID, containerID string,
+	body models.Channelable,
+) error {
+	_, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Patch(ctx, body, nil)
+
+	if err != nil {
+		return graph.Wrap(ctx, err, "patching event calendar")
+	}
+
+	return nil
+}
+
+// ---------------------------------------------------------------------------
+// message
+// ---------------------------------------------------------------------------
+
+// GetItem retrieves a Messageable item.
+func (c Channels) GetMessage(
+	ctx context.Context,
+	teamID, channelID, itemID string,
+	immutableIDs bool,
+	errs *fault.Bus,
+) (serialization.Parsable, *details.GroupsInfo, error) {
+	var (
+		size int64
+	)
+
+	// is preferImmutableIDs headers required here
+	message, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(channelID).
+		Messages().
+		ByChatMessageId(itemID).
+		Get(ctx, nil)
+	if err != nil {
+		return nil, nil, graph.Stack(ctx, err)
+	}
+
+	return message, MessageInfo(message, size), nil
+}
+
+func (c Channels) PostMessage(
```

Comment:
Hmm, do we need `PostMessage` and `PostReply`?  We won't be restoring these values yet, only exporting, which is the primary purpose of having these funcs.  I suppose the question is whether we need them for testing purposes (and delete, too)?  Maybe.  Maybe not.

I'd suggest that for this PR specifically you can drop both funcs.  Focus only on what's necessary for backup.  We can come back and re-add them again if necessary.